Makefile

```diff
@@ -704,32 +694,86 @@ DEFAULT_FLAGS = \
 # WASM distribution
 # -----------------
 #
-# Does something different and overrides pretty much everything.
+# We disable anything that is not pure Low*; no intrinsics; no EverCrypt
 
-# Should be fixed by having KreMLin better handle imported names
-WASM_STANDALONE=Prims LowStar.Endianness C.Endianness \
-  C.String TestLib C WasmSupport
+# TODO: the way externals are handled in Wasm is nuts and they should be in a
+# single module rather than require clients to do their surgical bundling.
+WASM_STANDALONE=Prims LowStar.Endianness C.Endianness C.String TestLib
 
-# Notes: only the functions reachable via Test.NoHeap are currently enabled.
 WASM_FLAGS	=\
   $(patsubst %,-bundle %,$(WASM_STANDALONE)) \
-  -no-prefix Test.NoHeap \
-  -bundle Test.NoHeap=Test,Test.* \
   -bundle FStar.* \
-  -bundle EverCrypt.*,Hacl.*,MerkleTree.*[rename=EverCrypt] \
   -bundle LowStar.* \
   -bundle Lib.RandomBuffer.System \
-  -bundle '\*[rename=Misc]' \
-  -minimal -wasm
-
-# Customizations for WASM.
-# - only keep definitions reachable from Test.NoHeap -- this indicates what we
-#   should retain for the WASM distribution.
-dist/wasm/Makefile.basic: TEST_FLAGS = -d wasm
+  -bundle Lib.Memzero \
+  -minimal -wasm -d wasm
+
+dist/wasm/Makefile.basic: VALE_ASMS =
+dist/wasm/Makefile.basic: HAND_WRITTEN_OPTIONAL_FILES =
+dist/wasm/Makefile.basic: HAND_WRITTEN_H_FILES =
+dist/wasm/Makefile.basic: HACL_OLD_FILES =
+dist/wasm/Makefile.basic: HAND_WRITTEN_FILES =
+dist/wasm/Makefile.basic: TARGETCONFIG_FLAGS =
+dist/wasm/Makefile.basic: INTRINSIC_FLAGS =
+
+# Must appear early on because of the left-to-right semantics of -bundle flags.
 dist/wasm/Makefile.basic: HAND_WRITTEN_LIB_FLAGS = $(WASM_FLAGS)
-dist/wasm/Makefile.basic: BUNDLE_FLAGS =
-dist/wasm/Makefile.basic: WASMSUPPORT_BUNDLE =
 
+# Overriding EverCrypt.Hash so that is it no longer a live root; it will be
+# eliminated via the -bundle EverCrypt.* below
+dist/wasm/Makefile.basic: E_HASH_BUNDLE =
+
+# Doesn't work in WASM because one function has some heap allocation
+dist/wasm/Makefile.basic: HASH_BUNDLE += -bundle Hacl.HMAC_DRBG
+
+# Doesn't work in WASM because un-materialized externals for AES128
+dist/wasm/Makefile.basic: FRODO_BUNDLE = -bundle Hacl.Frodo.KEM,Frodo.Params,Hacl.Impl.Frodo.*,Hacl.Impl.Matrix,Hacl.Frodo.*,Hacl.Keccak,Hacl.AES128
+
+# Doesn't work in Wasm because it uses assembler intrinsics
+ECDSA_BUNDLE = -bundle Hacl.Impl.ECDSA,Hacl.Impl.ECDSA,Hacl.Impl.ECDSA.*,Hacl.Impl.P256.*,Hacl.Impl.P256,Hacl.Spec.P256.*,Hacl.Impl.SolinasReduction,Hacl.Impl.LowLevel
+
+# No Vale Curve64; no "Local" Curve64 (local Makefile hack); only the slow 64
+# one
+dist/wasm/Makefile.basic: CURVE_BUNDLE = $(CURVE_BUNDLE_BASE) \
+	-bundle 'Hacl.Curve25519_64_Slow' \
```

Comment:
Ok, but actually this line 738 I added is not preventing the slow wasm curve from being generated...

Reply:
I guess the comment should read: no vale curve64, no local curve64, no slow 64 curve, only the 51 one -- line 738 *does* disable curve64/slow, correct? (curve64/slow = curve64 with intrinsics implement in Low*)

the only curve we're distributing for wasm is curve51

Reply:
You're right about the comment, I modified it so that it's clearer. I don't know what line 738 is doing, but `make dist/wasm/Makefile.basic` does produces a file named `Hacl_Curve25519_64_Slow.wasm`, even with line 738 enabled. I'll investigate with `-d reachability`

Reply:
ah good catch, Curve25519_64_Slow should indeed be eliminated... thanks for investigating

Reply:
`-d reachability` output : 
```
REACHABILITY: Hacl.Curve25519_64_Slow.ecdh is a root because it isn't private
```
I tried all `-bundle` combinations, impossible to prevent `Hacl_Curve25519_64_Slow.wasm`. However, putting `-drop Hacl.Curve25519_64.Slow` achieves the exact result I want in a single command. I know this option is deprecated but, here I don't know how else to do.

Reply:
Ok, this worked :
```
dist/wasm/Makefile.basic: CURVE_BUNDLE_SLOW =
```
Thanks @protz  for the debugging help !